### PR TITLE
add default componments

### PR DESCRIPTION
**Pull request by waruqi**
https://github.com/xmake-io/xmake/issues/3164#issuecomment-1346782590

### CODE REVIEW

File: xmake/core/project/target.lua [hunk 1]
```diff
@@ -416,9 +416,10 @@ function _instance:get_from_pkgs(name, opt)
         local configinfo = self:pkgconfig(pkg:name())
         -- get values from package components
         -- e.g. `add_packages("sfml", {components = {"graphics", "window"}})`
-        if configinfo and configinfo.components and pkg:components() then
+        local components_default = pkg:components_default()
+        if configinfo and (configinfo.components or components_default) and pkg:components() then
             local components_enabled = hashset.new()
-            for _, comp in ipairs(table.wrap(configinfo.components)) do
+            for _, comp in ipairs(table.join(configinfo.components or {}, components_default)) do
```
Comment (SirLynix) [inline xmake/core/project/target.lua:422]: doesn't that mean that those components can't be disabled? that's not what I meant sorry for the confusion.

`for _, comp in ipairs(table.wrap(configinfo.components or components_default)) do` is more what I suggest

Comment (waruqi) [reply xmake/core/project/target.lua:422]: ? Components that are not set with {compoments=""} and default=true are disabled by default.

you can do not set them.

Comment (waruqi) [reply xmake/core/project/target.lua]: I have improved it, try it again.

File: xmake/core/project/target.lua [hunk 2]
```diff
@@ -416,9 +416,10 @@ function _instance:get_from_pkgs(name, opt)
         local configinfo = self:pkgconfig(pkg:name())
         -- get values from package components
         -- e.g. `add_packages("sfml", {components = {"graphics", "window"}})`
-        if configinfo and configinfo.components and pkg:components() then
+        local components_default = pkg:components_default()
+        if configinfo and (configinfo.components or components_default) and pkg:components() then
```
Comment (SirLynix) [inline xmake/core/project/target.lua:420]: this won't work if configinfo is nil (which is the case with `add_packages("xx")`

this works:
```lua
        local selected_components = configinfo and configinfo.components or pkg:components_default()
        if selected_components and pkg:components() then
            local components_enabled = hashset.new()
            for _, comp in ipairs(table.wrap(selected_components)) do
```

Comment (waruqi) [reply xmake/core/project/target.lua]: try it again.

Comment (SirLynix) [reply xmake/core/project/target.lua]: works fine, thanks! 👍

**Pull request merged**